handler.go

```diff
@@ -95,14 +97,21 @@ func (h *Handler) Provision(ctx caddy.Context) error {
 				finalReplace := placeholderRepl.ReplaceKnown(repl.Replace, "")
 
 				if repl.re != nil {
-					tr := replace.RegexpString(repl.re, finalReplace)
+					tr := replace.RegexpIndexFunc(repl.re, func(src []byte, index []int) []byte {
+						template := h.ReplacePlaceholders(finalReplace)
+						return repl.re.Expand(nil, []byte(template), src, index)
+					})
 
 					// See: https://github.com/icholy/replace/issues/5#issuecomment-949757616
 					tr.MaxMatchSize = 2048
 					transforms[i] = tr
 				} else {
 					finalSearch := placeholderRepl.ReplaceKnown(repl.Search, "")
-					transforms[i] = replace.String(finalSearch, finalReplace)
+					// This would require the following method in the upstream icholy/replace/replace.go
+					// func StringWithLateBinding(old, new string, binding func(string) string) Transformer {
+					// 	return Bytes([]byte(binding(old)), []byte(binding(new)))
+					// }
+					transforms[i] = replace.StringWithLateBinding(finalSearch, finalReplace, h.ReplacePlaceholders)
```

Comment:
```suggestion
					transforms[i] = replace.String(
						h.ReplacePlaceholders(finalSearch),
						h.ReplacePlaceholders(finalReplace),
					)
```

^ this is functionally equivalent to the code you're proposing. Ie, the code you're proposing doesn't actually do late binding.

Reply:
@icholy You rock! I tested your suggested code and they all work like a champ!
@mholt I closed all previous PRs and created a new one using code that are suggested by @icholy. Please review.
P.S. I've learned a lot about the super dynamic nature of Transformer itself.

Reply:
Interesting! Thank you @icholy -- really appreciate that you spared a few minutes of your expertise here.